### ISSUE TITLE
false unreachable error

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#1746]
### Description

Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&reportUnannotatedClassAttribute=false&code=GYJw9gtgBAxmB2AXApgD0QGwJYCMpYgAcwREoBRVLRAZUQEMYBrAGigGcBXQwkZd9gChhAE2TAowABQBKAFzCoSqAHdqACwpVaDZgG0cYMBgC6sqPXYddTBcvvXGTAHTIkyEAH0479FK48fAJSAEqcSATI5CDgIDIygg7KIPRY7MhQYREQUTEkUgBEwFgg7IgFCYnKvFhIUgCMlfZqiJqU1HRO5paOzHZJvS5uKF4%2BKH4BvPzsoeGIkdGx8VVJKWkZWfM5i-kF6T4iFYrVILWIUgBMlcKC0glAA)

```python
from contextlib import ExitStack, suppress


def f():

    with ExitStack[bool]() as stack:
        stack.enter_context(suppress(RuntimeError))
        raise RuntimeError("first")

    print(1)

    with ExitStack() as stack:
        stack.enter_context(suppress(RuntimeError))
        raise RuntimeError("second")

    print(2)  # reported unreachable, but prints at runtime




f()
```
